### PR TITLE
chore(analytics)!: Remove Analytics Prefix

### PR DESCRIPTION
**Pull request by fjnoyp**
Remove the analytics prefix for userProfile / properties / location.

This is to enable better cross plugin usage of these types with Push Notifications.

The types will still exist in Analytics folder but in future work these types may be moved to a separate folder that is used by Analytics and Push Notifications categories.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by fjnoyp** [comment]
Moving to Draft - need to confirm approach in a design review meeting

**Comment by fjnoyp** [comment]
Marking as ready for review.  Teams already agreed on changes and will get final approval today hopefully at 1 pm PST

File: packages/amplify_core/lib/src/category/amplify_analytics_category.dart
Comment (HuiSF) [inline packages/amplify_core/lib/src/category/amplify_analytics_category.dart:72]: nit: trailing comma

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart
Comment (dnys1) [inline packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart:20]:
```suggestion
class UserProfile {
  const UserProfile({
    this.name,
    this.email,
    this.plan,
    this.location,
    this.customProperties,
  });
  
  final String? name;
  final String? email;
  final String? plan;
  final UserProfileLocation? location;
  final CustomProperties? customProperties;
```

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart
Comment (HuiSF) [inline packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart:7]: The linter rule has not been enabled for core, but we should add documentation.

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart
Comment (dnys1) [inline packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart:4]: Same comment as above

File: packages/amplify_core/lib/src/types/analytics/analytics/custom_properties.dart
Comment (HuiSF) [inline packages/amplify_core/lib/src/types/analytics/analytics/custom_properties.dart:6]: Documentation

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart
Comment (HuiSF) [inline packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart:4]: Ditto

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart
Comment (dnys1) [inline packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart:7]: Add to file

```dart
@Deprecated('Use UserProfile instead')
typedef AnalyticsUserProfile = UserProfile;
```

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart
Comment (dnys1) [reply packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart:4]: Add to file

```dart
@Deprecated('Use UserProfileLocation instead')
typedef AnalyticsUserProfileLocation = UserProfileLocation;
```

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart
Comment (dnys1) [reply packages/amplify_core/lib/src/types/analytics/analytics/user_profile.dart:20]: Add deprecation for `properties` field as well

```dart
  @Deprecated('Use customProperties instead')
  CustomProperties? get properties => customProperties;
```

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart
Comment (HuiSF) [reply packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart:4]: What's the plan communicate the migration plan? i.e. `remove "Analytics" prefix`?

Also `customEvent.properties` => `customEvent.customProperties`.

File: packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/aws_pinpoint_user_profile.dart
Comment (HuiSF) [inline packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/aws_pinpoint_user_profile.dart:22]: This new type is for aligning with amplify-swift correct? `userAttributes` is nullable in  amplify-swift though

File: packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart
Comment (fjnoyp) [reply packages/amplify_core/lib/src/types/analytics/analytics/user_profile_location.dart:4]: Yes I will need to update the documentation.  My understanding is that Dillon will change docs to allow v1 vs. v2 docs.  Once that is merged I will add a docs PR to include a warning for these breaking changes.

File: packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/aws_pinpoint_user_profile.dart
Comment (fjnoyp) [reply packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/aws_pinpoint_user_profile.dart:22]: Thanks good catch.  Idea behind non nullable was since you're using the Pinpoint subtype you should be forced to actually use that field, but would prefer to be consistent with iOS here.  And underlying data representation technically doesn't require it anyway.

**Review by HuiSF — APPROVED** [review]

File: packages/amplify_core/lib/src/types/analytics/analytics/custom_properties.dart
Comment (dnys1) [inline packages/amplify_core/lib/src/types/analytics/analytics/custom_properties.dart:7]: This needs a typedef with deprecation message

**Review by dnys1 — APPROVED** [review]

**Pull request merged**
